src/uniswap/incentives.ts

```diff
@@ -412,7 +460,53 @@ const checkTimeRanges = (
   }
 };
 
+export interface IncetivesDistributionReportAsJson {
+  from: Date;
+  to: Date;
+  dustLevel: number;
+  incentives: Array<[string, number]>;
+  incentivesTotal: number;
+}
+
+export const printIncentivesDistributionAsJson = (
+  out: Console,
+  distributions: IncentivesDistribution,
+  incentivesDustLevel: number
+) => {
+  const { from, to, incentivesTotal, providers } = distributions;
+
+  const providerAddresses = Object.keys(providers);
+
+  providerAddresses.sort((addr1, addr2) => {
```

Comment:
I must admit, I have hard time using lodash at times.
Probably, because of all the complexity it uses, trying to support several different styles.
It does not have simple types that I can read.

In this case, the code you suggested did not compile, and it took me some time to figure out what is wrong.
The shape of the data changes from one function to the next.
The types do not help, as they are overly generic.
Documentation is overly simplified, without any specific details, so did not help either.
LSP is incapable of showing specific types, so I can not follow the shape changes.

Maybe there are libraries that provide a similar toolset, but use just one specific approach?
So that they can have one type, instead of many complicated overloads.

TypeScript added a problem as well, as it did not infer the right type.
It inferred `[string | number]` for the return type of the lambda inside `map`, instead of `[string, number]`.
Here is what compiled:

```typescript
  const incentives = _(providers)
    .pickBy(({ incentives }) => incentives > incentivesDustLevel)
    .map(({ incentives }, address): [string, number] => [address, incentives])
    .sortBy(([address, _incentives]) => address.toLowerCase())
    .value();
```
---

> In general, maybe putting calculations and output in a single function is not the best idea? How about it to return an object and the upper code would make the decision on sending the sirialized object to console.log?

I would not say that what is happening here is a "calculation".
The way I see it is that it is filtering and sorting, that is needed for this particular output.
So, seems reasonable to keep them together.